### DISCUSSION TITLE
Modifying inducing points tensor for InducingPointKernel

### DISCUSSION THREAD

**Discussion by archielee** [cornellius-gp/gpytorch#1702] [Q&A] [answered]
Hi,

I've been using GPyTorch in the context of a growing dataset. With `ExactGP`s, adapting to the increased number of datapoints was quite simple (use `set_train_data()` to update the data in the GP, then re-optimize the hyperparameters). I was hoping to use sparse GPR to improve scalability.

I've been using the [Sparse GPR example](https://docs.gpytorch.ai/en/v1.5.0/examples/02_Scalable_Exact_GPs/SGPR_Regression_CUDA.html) in the docs as a template, which does not address increasing data or changing the inducing points. Is it a simple matter of changing the data in the `inducing_points` parameter within [`InducingPointKernel`](https://github.com/cornellius-gp/gpytorch/blob/v1.5.0/gpytorch/kernels/inducing_point_kernel.py#L25) and then re-optimizing? Is there some cache that needs to be cleared?

Thanks in advance for the help!

**Comment by wjmaddox** [comment, marked as answer]
Online variational inference (with GPs) is a bit of an open research question :) but yes, you should be able to get some mileage out of just updating the `inducing_points` parameter and then setting the model to be in train mode to clear the caches.

**Reply by archielee** [reply]
Alright, I'll give that a go. Thanks for the tip!
